### PR TITLE
fix(agw): Added cleanups for flaky retry support

### PR DESCRIPTION
**Pull request by VinashakAnkitAman**
## Title
fix(agw): Added cleanups for flaky retry support

## Summary
This PR adds some additional cleanups which were needed for rerunning a test case to observe its flaky behaviour as part of the github ssue #10596

## Test plan
Verified with sanity

Signed-off-by: VinashakAnkitAman <ankit.aman@radisys.com>

### CODE REVIEW

File: lte/gateway/python/integ_tests/s1aptests/s1ap_wrapper.py [hunk 1]
```diff
@@ -406,93 +436,132 @@ def ue_req(self):
...
         return self._trf_util
 
     @property
     def magmad_util(self):
+        """Get magmad_util instance"""
         return self._magmad_util
 
-    def cleanup(self):
+    @classmethod
+    def get_test_status(cls, test):
```
Comment (ardzoht) [inline lte/gateway/python/integ_tests/s1aptests/s1ap_wrapper.py]:
```suggestion
    def get_test_status(cls, test) -> boolean:
```

Comment (ardzoht) [reply lte/gateway/python/integ_tests/s1aptests/s1ap_wrapper.py]: maybe `is_test_successful` is a better naming for this function? (boolean return)

File: lte/gateway/python/integ_tests/s1aptests/s1ap_wrapper.py [hunk 2]
```diff
@@ -406,93 +436,132 @@ def ue_req(self):
...
 
+        if not is_test_passed:
+            print("The test has failed. Restarting Sctpd for cleanup")
+            self.magmad_util.restart_sctpd()
+            self.magmad_util.print_redis_state()
+
         # Cloud cleanup needs to happen after cleanup for
         # subscriber util and mobility util
         # if self._test_cloud:
         #    self._cloud_manager.clean_up()
```
Comment (ardzoht) [inline lte/gateway/python/integ_tests/s1aptests/s1ap_wrapper.py]: looks like dead code, can we remove it?

File: lte/gateway/python/integ_tests/s1aptests/s1ap_wrapper.py [hunk 3]
```diff
@@ -406,93 +436,132 @@ def ue_req(self):
...
+    @classmethod
+    def get_test_status(cls, test):
+        """Get current test case execution status"""
+        if test is None:
+            test = inspect.currentframe().f_back.f_back.f_locals["self"]
+        if test is not None and hasattr(test, "_outcome"):
+            result = test.defaultTestResult()
+            test._feedErrorsToResult(result, test._outcome.errors)
+            error = result.errors and result.errors[-1][0] is test
+            failure = result.failures and result.failures[-1][0] is test
```
Comment (ardzoht) [inline lte/gateway/python/integ_tests/s1aptests/s1ap_wrapper.py]:
```suggestion
            test_contains_error = result.errors and result.errors[-1][0] is test
            test_contains_failure = result.failures and result.failures[-1][0] is test
```

**Review by ardzoht — CHANGES_REQUESTED** [review]

**Review by ardzoht — APPROVED** [review]
lgtm

**Comment by ardzoht** [comment]
@VinashakAnkitAman python format check is failing with:
```
Running isort on lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py...
Running autopep8 on lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py...
Running add-trailing-comma on lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py...
Rewriting lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py
```

I'd recommend running https://github.com/magma/magma/tree/master/lte/gateway/docker/python-precommit for this

File: lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py [hunk 4]
```diff
@@ -1789,11 +1804,23 @@ def get_proxy_port_no(self) -> Optional[int]:
...
     # this adds similar API using `ovs-ofctl` cmd
     def get_flows(self, table_id) -> []:
+        """Fetch the OVS flow rules"""
         output = self.magma_utils.exec_command_output(
-            "sudo ovs-ofctl dump-flows gtp_br0 table={}".format(table_id),
+            "sudo ovs-ofctl dump-flows gtp_br0 table={0}".format(table_id),
         )
         return output.split("\n")
 
+    def delete_flows(self, table_id):
```
Comment (pruthvihebbani) [inline lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py:1909]: What is the difference between this function and delete_ovs_flow_rules?

Comment (VinashakAnkitAman) [reply lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py:1909]: delete_ovs_flow_rules function is used to expose the delete_flows function under gtp bridge utility to s1ap utillity class similar to other functions. All the raw ovs flow rules operations are wrapped under functions in gtp bridge utility class.

**Comment by pruthvihebbani** [comment]
LGTM. Python format check is failing. Can you fix that?

**Pull request merged**